dreadnode/main.py

```diff
@@ -163,6 +179,121 @@ def _get_profile_api_key(self, profile: str | None = None) -> str | None:
+        if self.organization:
+            self._organization = self._api.get_organization(self.organization)
+            if not self._organization:
+                raise RuntimeError(f"Organization '{self.organization}' not found.")
+
+        else:
+            organizations = self._api.list_organizations()
+
+            if not organizations:
+                raise RuntimeError(
+                    f"You are not part of any organizations on {self.server}. You will not be able to use Strikes.",
+                )
+
+            if len(organizations) > 1:
+                # We should not presume to choose an organization
+                org_list = "\t\n".join([f"- {o.name}" for o in organizations])
+                raise RuntimeError(
+                    f"You are part of multiple organizations. Please specify an organization from:\n{org_list}"
+                )
+            self._organization = organizations[0]
+
+    def _create_workspace(self, name: str) -> Workspace:
+        if self._api is None:
+            raise RuntimeError("API client is not initialized.")
+
+        try:
+            return self._api.create_workspace(name=name, organization_id=self._organization.id)
+        except RuntimeError as e:
+            if "403: Forbidden" in str(e):
+                raise RuntimeError(
+                    "You do not have permission to create the specified workspace for this organization. (You must be a Contributor to create new workspaces)."
+                ) from e
+            raise
+
+    def _resolve_workspace(self) -> None:
+        if self._api is None:
+            return
+
+        found_workspace: Workspace | None = None
+        if self.workspace:
+            try:
+                found_workspace = self._api.get_workspace(self.workspace)
+            except RuntimeError as e:
+                if "404: Workspace not found" in str(e):
+                    pass  # do nothing, we'll create it below
+                else:
+                    raise
+
+            if not found_workspace and isinstance(self.workspace, UUID):
+                raise RuntimeError(f"Workspace with ID '{self.workspace}' not found.")
+
+            if not found_workspace and isinstance(self.workspace, str):  # specified by name/slug
+                # create the workspace (must be an org contributor)
+                found_workspace = self._create_workspace(name=self.workspace)
+
+        else:  # the user provided no workspace, attempt to find a default one
+            workspaces = self._api.list_workspaces(
+                filters=WorkspaceFilter(
+                    org_id=self._organization.id if self._organization else None
+                )
+            )
+            default_workspace = next((ws for ws in workspaces if ws.is_default is True), None)
+            if default_workspace:
+                found_workspace = default_workspace
+            else:
+                found_workspace = self._create_workspace(name=DEFAULT_WORKSPACE_NAME)
```

Comment:
consider not auto-create if no default workspace found. The API handles this on org join by invitation or first user.

Reply:
maybe raise an error:

```python
else:
        raise RuntimeError(
            "No default workspace found. This shouldn't happen - please contact org owner."
        )